app/presenters/pages_presenter.rb

```diff
@@ -0,0 +1,41 @@
+class PagesPresenter
+
+  def initialize(q, start_date = nil, end_date = nil, verb_phrase = nil, noun_phrase = nil, sort_key = "feedback_comments", sort_dir = "desc")
+    @q = q
+    @start_date = start_date ||= DateTime.now.beginning_of_day - 30.days
+    @end_date = end_date ||= DateTime.now.end_of_day
+    @verb_phrase = verb_phrase
+    @noun_phrase = noun_phrase
+    @sort_key = sort_key if %w(pages.base_path feedback_comments).include?(sort_key)
+    @sort_dir = sort_dir if %w(asc desc).include?(sort_dir)
+  end
+
+  def rows
+    results.map do |page|
+      [
+        { text: page[:base_path] },
+        { text: page[:survey_count], format: "numeric" },
+        { text: Page.top_phrases(page[:page_id]).take(3).join(", ") },
```

Comment:
It doesn't feel right that the presenter needs to make a DB query here. I can see how it'd be easier to do this rather than pass a list of items _into_ the presenter here, but I'd keep the DB querying to the controllers.

Reply:
👍